state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1802,11 +1825,16 @@ pub mod pallet {
 			})
 		}
 
-		pub(super) fn take_network_fee(input: AssetAmount) -> FeeTaken {
+		pub(super) fn take_network_fee(input: AssetAmount, min_fee_enforced: bool) -> FeeTaken {
 			if input.is_zero() {
 				return FeeTaken { remaining_amount: 0, fee: 0 };
 			}
-			let (remaining, fee) = utilities::calculate_network_fee(T::NetworkFee::get(), input);
+
+			let min_fee = if min_fee_enforced { MinimumNetworkFeePerChunk::<T>::get() } else { 0 };
+
+			let (remaining, fee) =
+				utilities::calculate_network_fee(T::NetworkFee::get(), min_fee, input);
```

Comment:
Since this is called inside a loop, and the min fee/rate are the same on each iteration, it might make sense to refactor this method such that we pass the fee rate and min fee by value instead of reading storage.

Reply:
I believe storage items are cached, and reading the same value a few times in a row should be cheap, no? At the same time, if we move this outside of the loop, then we would be reading the storage even when it is not needed. Seems to me like it is not really worth it?

Reply:
They are cached in the storage overlay - ie. it doesn't require a full read/traversal of the underlying trie, but it still needs to hash the key, look it up in the backend k-v store and decode the result. It's wasteful but I agree it's not a big deal, feel free to ignore.